skyportal/handlers/api/observation.py

```diff
@@ -0,0 +1,455 @@
+
+            cum_prob = (
+                sa.func.sum(
+                    LocalizationTile.probdensity * LocalizationTile.healpix.area
+                )
+                .over(order_by=LocalizationTile.probdensity.desc())
+                .label('cum_prob')
+            )
+            localizationtile_subquery = (
+                sa.select(LocalizationTile.probdensity, cum_prob).filter(
+                    LocalizationTile.localization_id == localization.id
+                )
+            ).subquery()
+
+            min_probdensity = (
+                sa.select(
+                    sa.func.min(localizationtile_subquery.columns.probdensity)
+                ).filter(
+                    localizationtile_subquery.columns.cum_prob <= localization_cumprob
+                )
+            ).scalar_subquery()
+
+            tiles_subquery = (
+                sa.select(InstrumentField.id)
+                .filter(
+                    LocalizationTile.localization_id == localization.id,
+                    LocalizationTile.probdensity >= min_probdensity,
+                    InstrumentFieldTile.instrument_id == instrument.id,
+                    InstrumentFieldTile.instrument_field_id == InstrumentField.id,
+                    InstrumentFieldTile.healpix.overlaps(LocalizationTile.healpix),
+                )
+                .subquery()
+            )
+
+            obs_query = obs_query.join(
+                tiles_subquery,
+                ExecutedObservation.instrument_field_id == tiles_subquery.c.id,
+            )
+
+            if return_probability:
+                union = (
+                    sa.select(
+                        ha.func.union(InstrumentFieldTile.healpix).label('healpix')
+                    )
+                    .filter(
+                        InstrumentFieldTile.instrument_id == instrument.id,
+                        InstrumentFieldTile.instrument_field_id
+                        == ExecutedObservation.instrument_field_id,
+                        ExecutedObservation.obstime >= start_date,
+                        ExecutedObservation.obstime <= end_date,
+                    )
+                    .subquery()
+                )
+                prob = sa.func.sum(
+                    LocalizationTile.probdensity
+                    * (union.columns.healpix * LocalizationTile.healpix).area
+                )
+                query = sa.select(prob).filter(
+                    LocalizationTile.localization_id == localization.id,
+                    union.columns.healpix.overlaps(LocalizationTile.healpix),
+                )
+                intprob = DBSession().execute(query).scalar_one()
+
+        observations = obs_query.all()
+
+        if return_probability:
+            data = {
+                "observations": [o.to_dict() for o in observations],
+                "probability": intprob,
+            }
+        else:
+            data = observations
```

Comment:
You don't need to dictionarify it here?

Reply:
It should automatically be the handler right?